<launch>
    <arg name="leftBagPath" default="empty/dir"/>
    <arg name="rightBagPath" default="empty/dir"/>
    <arg name="leftBagTopic" default="unfiltered_radar_packet"/>
    <arg name="rightBagTopic" default="unfiltered_radar_packet"/>
    <arg name="leftScanMode" default="0"/>
    <arg name="rightScanMode" default="0"/>

    <group ns="radar_right">
        <node pkg="rosbag" type="play" name="rosbag" 
            args="-l $(arg rightBagPath)"/>
        <node name="radar_processor" pkg="radar_driver" type="radar_processor"
            args="-s $(arg rightScanMode) -t $(arg rightBagTopic)"/>   
        <node name="radar_visualizer" pkg="radar_driver" type="radar_visualizer"
            args="-t filtered_radar_packet -n radar_right"/>  
    </group>

    <group ns="radar_left">
        <node pkg="rosbag" type="play" name="rosbag" 
            args="-l $(arg leftBagPath)"/>
        <node name="radar_processor" pkg="radar_driver" type="radar_processor"
            args="-s $(arg leftScanMode) -t $(arg leftBagTopic)"/> 
        <node name="radar_visualizer" pkg="radar_driver" type="radar_visualizer"
            args="-t filtered_radar_packet -n radar_left"/> 
    </group>

    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find radar_driver)/rviz/radarPointCloud.rviz" />
    
</launch>
